src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1324,6 +1325,59 @@ def create_or_replace_view(
             source_plan,
         )
 
+    def find_table_function_in_sql_tree(self, plan: SnowflakePlan):
+        """This function is meant to find any udtf function call from a create dynamic table plan and
+        replace '*' with explicit column identifier in the select of table function. Since we cannot
+        differentiate udtf call from other table functions, we apply this change to all table functions.
+        """
+        deepcopied_plan = copy.deepcopy(plan)
+        queue = deque()
+        queue.append(deepcopied_plan)
+        from snowflake.snowpark._internal.analyzer.select_statement import (
+            SelectTableFunction,
+            Selectable,
+        )
+
+        while queue:
+            deepcopied_plan = queue.popleft()
+            for node in deepcopied_plan.children_plan_nodes:
+                queue.append(node)
+
+            # the bug only happen when create dynamic table on top of a table function
+            # this is meant to decide whether the plan is select from a tale function
+            if isinstance(deepcopied_plan, SelectTableFunction) and isinstance(
+                deepcopied_plan.snowflake_plan.source_plan, TableFunctionJoin
+            ):
+                if (
+                    deepcopied_plan.snowflake_plan.source_plan.right_cols == ["*"]
+                    and len(deepcopied_plan.snowflake_plan.children_plan_nodes) == 1
+                ):
+                    child_plan = deepcopied_plan.snowflake_plan.children_plan_nodes[0]
+                    if isinstance(child_plan, Selectable):
+                        child_plan = child_plan.snowflake_plan
+                    deepcopied_plan.snowflake_plan.source_plan.right_cols = (
+                        deepcopied_plan.snowflake_plan.quoted_identifiers[
+                            len(child_plan.quoted_identifiers) :
+                        ]
+                    )
+                    new_plan = self.session._analyzer.resolve(
+                        deepcopied_plan.snowflake_plan.source_plan
+                    )
+                    deepcopied_plan._snowflake_plan = new_plan
+
+                    # resolve the plan to apply change
+                    self.session._analyzer.resolve(
+                        deepcopied_plan.snowflake_plan.source_plan  # type: ignore
+                        if isinstance(deepcopied_plan, Selectable)
```

Comment:
is is possible for a `Selebtable` to have a `snowflake_plan` where `source_plan is None`. Can we make sure that is not the case here?

Reply:
I think the case shall not happen here, @sfc-gh-jdu  can you help me confirm?